lib/uplink/bucket.go

```diff
@@ -0,0 +1,73 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package uplink
+
+import (
+	"context"
+	"io"
+	"time"
+
+	"storj.io/storj/pkg/storj"
+)
+
+// Bucket represents operations you can perform on a bucket
+type Bucket struct {
+	Access *Access
+}
+
+// Get returns an Object, if authorized.
+func (b *Bucket) Get(ctx context.Context, bucket string, path storj.Path) (*Object, error) {
+	panic("TODO")
+}
+
+// UploadOpts controls options about uploading a new Object, if authorized.
+type UploadOpts struct {
+	Metadata map[string]string
+	Expires  time.Time
+
+	// EncryptionScheme determines the object's encryption scheme. If not set, uses the Uplink default
+	EncryptionScheme *storj.EncryptionScheme
+}
+
+// Upload uploads a new object, if authorized.
+func (b *Bucket) Upload(ctx context.Context, bucket string, path storj.Path, data io.Reader, opts UploadOpts) error {
+	panic("TODO")
+}
+
+// Delete removes an object, if authorized.
+func (b *Bucket) Delete(ctx context.Context, bucket string, path storj.Path) error {
```

Comment:
Perhaps the name should be `DeleteObject`, so it is clear that the method is deleting an object from the current bucket, instead of deleting the current bucket itself.

Reply:
yeah good call